### PR TITLE
Layer refactor

### PR DESCRIPTION
**Pull request by MrPerson**
- Removed level var
- Removed the showpipe var on pipes.
- Removed all layer and level definitions on all the maps. Changed a hidden pipe in the incinerator to be a visible one. Sorry to anyone working meta/mini but this shit is needed to make it compile.
- Moved all layer definitions to the defines folder. These are all up in the air and can be changed easily.
- Changed up cables, pipes, and other below-the-floor things to use regular icon states and have a generic hide() proc instead of using a "hidden" icon state when below the floor that the t-ray reveals and redefining hide() for everything that needed that functionality. Much cleaner!
- Cables. pipes, disposal tubes with procs like update() and updateicon() use update_icon() instead.
- Added a new invisibility level, INVISIBILITY_UNDER_THE_FLOOR
- Made every initialize() call its parent because they should have been anyways. Atmospheric machinery calls update_icon() in its default initialize() instead of having every machine call update_icon() independently.
- T-ray scanners use animate() instead of special icon_states.
- Removed the... thing?... in code/game/machinery/Beacon.dm. If you can tell me what it is I'll fix it up I guess. I don't think it was working anyways, whatever it was.
- A bunch of changes to atmospherics machinery to put all the pipe-based underlays onto the same layer as whatever node the machine is connecting to OR the pipe layer if there is no node.
- Turfs are at a very, very low layer and then get set to their correct one as part of New(). This is so mappers can see cables and pipes while mapping.
- Changed the icons of pumps and volume pumps to make them look like how they used to look before the red band was placed on the pipe. I only did this because that red band would be floating in the air if the pump's pipe connection was under the floor, which looked awful.

Most of this pull request is replacing layer definitions with defines or the map changes.

Pictures:

### CODE REVIEW

**Comment by Miauw62** [comment]
Holy shit that animation is sexy.
I do spot it looking a bit weird with the scrubber though.

**Comment by RemieRichards** [comment]
I replaced your picture link with markdown.

**Comment by optimumtact** [comment]
The connector sprites in the pipes seem to be shaded in a darker colour, or they fade out at a darker colour.

File: code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm [hunk 1]
```diff
@@ -55,13 +55,13 @@ Iconnery
 	var/unconnected = (~connected) & initialize_directions
 	for(var/direction in cardinal)
 		if(unconnected & direction)
-			overlays += getpipeimage('icons/obj/atmospherics/trinary_devices.dmi', "intact", direction)
+			underlays += getpipeimage('icons/obj/atmospherics/trinary_devices.dmi', "broken", direction, , PIPE_LAYER)
```
Comment (optimumtact) [inline code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm]: Why change from intact to broken?

**Comment by Jalleo** [comment]
The machinery Beacon is meant to be a anchored version of the normal teleporter beacon. I think what it was meaning was basically for it to be one which you can activate and leave.

**Comment by Cheridan** [comment]
>Removed the... thing?... in code/game/machinery/Beacon.dm. If you can tell me what it is I'll fix it up I guess. I don't think it was working anyways, whatever it was.

It's basically just a teleporter beacon, but instead of being a mobile item it's an underfloor structure. It was used on the old tcomm satellite.

**Comment by MrPerson** [comment]
Do you think it's worth keeping? Either way I guess I need to add an actual beacon there to maintain existing functionality.

**Comment by Jalleo** [comment]
There are a few places where a beacon should be on the floor just to make things easier in moving about. e.g the AI sat, bridge,
If we then require it to have power it would create a new dynamic but that would require a rework a bit would be for another PR at somepoint

**Comment by Cheridan** [comment]
I'm not sure if the floor-beacon is even in use now that the tcomm sat is derelict. It would be fine to ditch, I think.

**Pull request closed without merging**